network/wsNetwork.go

```diff
@@ -1956,6 +1978,57 @@ func (wn *WebsocketNetwork) prioWeightRefresh() {
 	}
 }
 
+func primaryNetwork(network protocol.NetworkID) string {
+	return fmt.Sprintf("algorand-%s.network", network)
+}
+
+func secondaryNetwork(network protocol.NetworkID) string {
+	return fmt.Sprintf("algorand-%s.net", network)
+}
+
+var networkIDToPrimarySRVBaseURIRegex = map[protocol.NetworkID]*regexp.Regexp{
+	config.Mainnet: regexp.MustCompile(primaryNetwork(config.Mainnet)),
+	config.Testnet: regexp.MustCompile(primaryNetwork(config.Testnet)),
+}
+
+var networkIDToSecondarySRVBaseURIRegex = map[protocol.NetworkID]*regexp.Regexp{
+	config.Mainnet: regexp.MustCompile(secondaryNetwork(config.Mainnet)),
+	config.Testnet: regexp.MustCompile(secondaryNetwork(config.Testnet)),
+}
+
+func (wn *WebsocketNetwork) mergePrimarySecondaryRelayAddressLists(network protocol.NetworkID,
+	primaryRelayAddresses []string, secondaryRelayAddresses []string) (dedupedRelayAddresses []string) {
+	// Initialize w/ worst case capacity
+	//dedupedRelayAddresses = make([]string, 0, len(primaryRelayAddresses)+len(secondaryRelayAddresses))
+	//copy(dedupedRelayAddresses, primaryRelayAddresses)
+
+	var relayAddressPrefixToValue = make(map[string]string, 2*len(primaryRelayAddresses))
+	var prgx = networkIDToPrimarySRVBaseURIRegex[network]
+
+	for _, pra := range primaryRelayAddresses {
+		var normalizedPra = strings.ToLower(pra)
+		relayAddressPrefixToValue[prgx.ReplaceAllString(normalizedPra, "")] = normalizedPra
+	}
+
+	var srgx = networkIDToSecondarySRVBaseURIRegex[network]
+
+	for _, sra := range secondaryRelayAddresses {
+		var normalizedSra = strings.ToLower(sra)
+		var pfxKey = srgx.ReplaceAllString(normalizedSra, "")
+		// Add to our address map if prefix does not exist, otherwise take no action (already covered by primary)
+		if relayAddressPrefixToValue[pfxKey] == "" {
+			relayAddressPrefixToValue[pfxKey] = normalizedSra
+		}
+	}
+
+	dedupedRelayAddresses = make([]string, 0, len(relayAddressPrefixToValue))
+	for _, value := range relayAddressPrefixToValue {
+		dedupedRelayAddresses = append(dedupedRelayAddresses, value)
```

Comment:
The value in a range may reuse the same object for every result, if you're saving the value like you have here it causes the same object to be added multiple times. I don't think it can do that with strings, but to be safe you can use this approach:
```suggestion
	for i := range relayAddressPrefixToValue {
		dedupedRelayAddresses = append(dedupedRelayAddresses, relayAddressPrefixToValue[i])
```

Reply:
I think you might be referring to the risk of two keys pointing to the same value. The point of this code is to de-dup the values with a twist (treating primary/secondary network suffixes as if they are the same for deduping).